### PR TITLE
feature: forta local development capability

### PR DESCRIPTION
**Pull request by sergeyWh1te**
# Added:

* Set up Forta-local mode development
Screenshot 2024-03-11 at 17 02 11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,9 +40,33 @@ mainnet)
...
 
-[template](./template) - Template for bot creation. (Forta version)
+## Local development
 
-[template-v2](./template-v2) - Template for bot creation. (Sergey Wh1te's version)
+To launch the bot environment similar to the production environment, several services need to be set up:
+
+- Message bus https://nats.io/
+- [Forta-node](https://github.com/forta-network/forta-node/blob/master/cmd/node/nodecmd/nodecmd.go#L64) - Forta wrapper over the Ethereum-like node
+- [Forta-scanner](https://github.com/forta-network/forta-node/blob/master/cmd/node/nodecmd/nodecmd.go#L40) service communicating with forta-agent via gRPC
```
Comment (TheDZhon) [inline README.md:49]:
```suggestion
- [NATS](https://nats.io/) — Message bus 
- [Forta-node](https://github.com/forta-network/forta-node/blob/master/cmd/node/nodecmd/nodecmd.go#L64) — Forta wrapper over the Ethereum-like node
- [Forta-scanner](https://github.com/forta-network/forta-node/blob/master/cmd/node/nodecmd/nodecmd.go#L40) — service communicating with forta-agent via gRPC
```

File: README.md [hunk 2]
```diff
@@ -40,9 +40,33 @@ mainnet)
...
+
+### Setup
+
+1. Install docker
+2. Clone forta-node repo to somewhere. `git clone https://github.com/forta-network/forta-node`
```
Comment (TheDZhon) [inline README.md:54]:
```suggestion
2. Clone forta-node repo: `git clone https://github.com/forta-network/forta-node`
```

File: README.md [hunk 3]
```diff
@@ -40,9 +40,33 @@ mainnet)
...
+3. `cd forta-node && make containers`. It creates containers for local using
```
Comment (TheDZhon) [inline README.md:55]:
```suggestion
3. `cd forta-node && make containers`. It creates containers for local usage
```

File: README.md [hunk 4]
```diff
@@ -40,9 +40,33 @@ mainnet)
...
+4. Go back to `alerting-forta` and provide your own `scan.jsonRpc.url` in forta-local-config.yml
+5. Run `docker-compose up -d`
+6. Run your service in production mode. For example [ethereum-steth](..%2Fethereum-steth)
+   1. cd [ethereum-steth](..%2Fethereum-steth)
+   2. `yarn start:prod`
+7. **Extra:** If you want to send agent's findings into Telegram chat or discord chat then provide value in config:
+   `localMode.webhookUrl`: `http://localhost:5001/hook/<your slug name>`
+
+Full capabilities for running in localmode you can find here: https://docs.forta.network/en/latest/scanner-local-mode/.
+Also, it's usefully for understanding `forta-scanner` config file - check https://github.com/forta-network/forta-node/blob/master/config/config.go#L155
```
Comment (TheDZhon) [inline README.md:65]:
```suggestion
Also, the `forta-scanner` config file is useful for a deep dive — check the following: https://github.com/forta-network/forta-node/blob/master/config/config.go#L155
```

File: README.md [hunk 5]
```diff
@@ -40,9 +40,33 @@ mainnet)
...
+
+### Or use emulation prod environment for local development.
+
+Just move to bot directory and run `yarn start`. [Forta-sdk](https://github.com/forta-network/forta-bot-sdk/tree/master/sdk) provides emulation prod environment for simplifier local development
```
Comment (TheDZhon) [inline README.md:69]:
```suggestion
Just navigate to the bot directory and run `yarn start`. [Forta-sdk](https://github.com/forta-network/forta-bot-sdk/tree/master/sdk) provides emulation prod environment for a simplified local development
```

File: forta-local-config.yml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+chainId: 1
+
+scan:
+  jsonRpc:
+    url: https://mainnet.infura.io/v3/<your secret key>
+
+localMode:
+  enable: true
+  logToStdout: true
+  privateKeyHex: 984d28fde238c32d06420db29b7277f5fd24c61954d2da491bf7d1ef19996bd0
```
Comment (TheDZhon) [inline forta-local-config.yml:10]: is it safe to place here?

**Review by TheDZhon — CHANGES_REQUESTED** [review]
Thank you for spearheading the effort!

File: forta-local-config.yml (hunk 6 above)
Comment (sergeyWh1te) [reply forta-local-config.yml:10]: Yes. For local goals(our setup) - unfortunately it's only unused required parameter for forta-scanner.

**Review by TheDZhon — APPROVED** [review]

**Pull request merged**